astroquery/irsa/core.py

```diff
@@ -368,24 +372,25 @@ def _parse_result(self, response, verbose=False):
 
         # Check if results were returned
         if 'The catalog is not on the list' in content:
-            raise Exception("Catalog not found")
+            raise ValueError("Invalid Catalog specified")
 
         # Check that object name was not malformed
         if 'Either wrong or missing coordinate/object name' in content:
-            raise Exception("Malformed coordinate/object name")
+            raise ValueError("Malformed coordinate/object name")
 
         # Check to see that output table size limit hasn't been exceeded
         if 'Exceeding output table size limit' in content:
-            raise Exception("Exceeded output table size - reduce number "
-                            "of output columns and/or limit search area")
+            raise TableParseError("Exceeded output table size - reduce number "
+                                  "of output columns and/or limit search area")
 
         # Check to see that the query engine is working
         if 'SQLConnect failed' in content:
-            raise Exception("The IRSA server is currently down")
+            raise TimeoutError("The IRSA server is currently down")
 
         # Check that the results are not of length zero
         if len(content) == 0:
-            raise Exception("The IRSA server sent back an empty reply")
+            warnings.warn("The IRSA server sent back an empty reply",
+                          NoResultsWarning)
```

Comment:
This is great to change this to the warning, however, an empty table has to be returned now after the warning.

Reply:
Thanks @bsipocz. No warning class seems to describe this case perfectly, should I add a `EmptyResultWarning` class in the **exceptions file** that can take care of situations like this ?

Reply:
Also, the check for an empty table is done here https://github.com/tinumide/astroquery/blob/22a6f09b0381fae187f63de18813478d436d0508/astroquery/irsa/core.py#L407 and an empty table is returned afterwards. Should I still go ahead to return the empty table here as well ?

Reply:
No, I think the `NoResultsWarning` is perfectly suitable here, we are using it for similar situations.

And you're absolutely right, the votable machinery already takes care of handling the table parsing, so no need to manually return an empty table.